AAEmu.Game/Core/Managers/TaskManager2.cs

```diff
@@ -151,7 +151,18 @@ public bool Cancel(Task task)
 
         return res;
     }
-
+    public void RemoveTasks(Func<Task, bool> predicate)
+    {
+        // Take a snapshot of the current tasks to avoid modifying the collection while iterating.
+        foreach (var kvp in _queue.ToArray())
+        {
+            if (predicate(kvp.Value))
+            {
+                _queue.Remove(kvp.Key, out _);
+                ReleaseId(kvp.Key);
```

Comment:
Probably more correct from a multi-threading perspective:
```suggestion
                if (_queue.TryRemove(kvp))
                    ReleaseId(kvp.Key);
```